src/oncall/api/v0/roster_user.py

```diff
@@ -44,15 +45,49 @@ def on_delete(req, resp, team, roster, user):
     roster_id = cursor.fetchone()
     if roster_id is None:
         raise HTTPNotFound()
+
```

Comment:
This logic can be handled by the scheduler. In particular, the round-robin scheduler uses the calendar to guess the previous user if one has been deleted, so we don't need to do any schedule management in a roster_user action.